qcodes/instrument_drivers/oxford/MercuryiPS_VISA.py

```diff
@@ -299,8 +299,89 @@ def __init__(self, name: str, address: str, visalib=None,
                            get_cmd=self._get_ramp_rate,
                            set_cmd=self._set_ramp_rate)
 
+
+        self.add_parameter('theta_saferamp',
+						   label='Theta',
+						   unit='degrees',
+						   set_cmd=self._theta_saferamp_set,
+						   get_cmd=self._theta_saferamp_get)
+		
+        self.add_parameter('r_saferamp',
+						   label='B_Radius',
+						   unit='T',
+						   set_cmd=self._r_saferamp_set,
+						   get_cmd=self._r_saferamp_get)
+
+        self.add_parameter('r_simulramp',
+						   label='B_Radius',
+						   unit='T',
+						   set_cmd=self._r_simulramp_set,
+						   get_cmd=self._r_simulramp_get)
+
+        self.add_parameter('phi_saferamp',
+						   label='Phi',
+						   unit='degrees',
+						   set_cmd=self._phi_saferamp_set,
+						   get_cmd=self._phi_saferamp_get)
+
         self.connect_message()
 
+    def _theta_saferamp_set(self, target):
+        """
+		Set the theta target and ramp to it
+		"""
+        self.theta_target(target)
+        self.ramp(mode='safe')
+        self.theta_measured()
```

Comment:
what is this call needed for? in order to update the value of the parameter, i guess? how is it used by Loop?